Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1049,8 +1050,8 @@ void RenderGrid::placeItemsOnGrid(std::optional<LayoutUnit> availableLogicalWidt
     performAutoPlacement();
 
 #if ASSERT_ENABLED
-    for (auto* gridItem = currentGrid().orderIterator().first(); gridItem; gridItem = currentGrid().orderIterator().next()) {
-        if (currentGrid().orderIterator().shouldSkipChild(*gridItem))
+    for (auto* gridItem : iterator) {
```

Comment:
The old code was debugging the iterator from the beginning. Shouldn't we do the same here ?

Reply:
I made the OrderIterator::begin() implementation call first() on the returned begin iterator object. This way the user does not have to call first themselves to get the iterator to start at the beginning.

Reply:
My question was more related to the `currentGrid().orderIterator()` change. It returns a reference to a member iterator, so it's not a fresh one and thus correct. But I'm not sure we want to change this part though (extracting `currentGrid().orderIterator()` into a variable), it makes the code less obvious to me.

Reply:
Oh I see. Extracting currentGrid().orderIterator() is not necessary. I somewhat prefer it, but I can revert that since I see how it could make it less obvious.